sdk/src/transaction/sanitized.rs

```diff
@@ -43,10 +43,17 @@ pub trait AddressLoader {
     fn load_addresses(&self, lookups: &[MessageAddressTableLookup]) -> Result<LoadedAddresses>;
 }
 
-pub struct DisabledAddressLoader;
-impl AddressLoader for DisabledAddressLoader {
+pub enum SimpleAddressLoader {
+    Disabled,
+    Enabled(LoadedAddresses),
+}
+
+impl AddressLoader for SimpleAddressLoader {
     fn load_addresses(&self, _lookups: &[MessageAddressTableLookup]) -> Result<LoadedAddresses> {
-        Err(TransactionError::UnsupportedVersion)
+        match self {
+            Self::Disabled => Err(TransactionError::AddressLookupTableNotFound),
+            Self::Enabled(loaded_addresses) => Ok(loaded_addresses.clone()),
```

Comment:
is the clone necessary or would a ref suffice?

Reply:
I refactored the `AddressLoader` trait to remove this clone